codec-http/src/main/java/io/netty/handler/codec/spdy/SpdyFrameCodec.java

```diff
@@ -284,6 +324,18 @@ public void write(ChannelHandlerContext ctx, Object msg, ChannelPromise promise)
                     spdyWindowUpdateFrame.deltaWindowSize()
             );
             ctx.write(frame, promise);
+        } else if (msg instanceof SpdyUnknownFrame) {
+            SpdyUnknownFrame spdyUnknownFrame = (SpdyUnknownFrame) msg;
+            try {
+                frame = spdyFrameEncoder.encodeUnknownFrame(
+                        ctx.alloc(),
+                        spdyUnknownFrame.frameType(),
+                        spdyUnknownFrame.flags(),
+                        spdyUnknownFrame.content());
+                ctx.write(frame, promise);
+            } finally {
+                spdyUnknownFrame.release();
+            }
```

Comment:
should we change `SpdyDataFrame` to this `try finally` style too?

Reply:
We can do this in a follow up as this would also apply to 4.1